Fixes and improvements

nrader95 opened the pull request:
Basically what i already described in #178 plus readme update.

nrader95 committed:
Remove samples building from .sln (Fix for sln build on non-windows OS'es)
IDE support should still work for them on solution open as ActiveCfg entries were not removed

nrader95 committed:
Meaningful Entity.Get exception message

nrader95 committed:
Entity.IsAlive world.trim fix + new faster Entity.IsAliveVersion property

nrader95 committed:
Systems use entity buffering by default
Except for threading, for performance reasons

nrader95 committed:
HelpPage update

nrader95 commented:
Oh and, those changes are actually tested by me and my colleagues for some time, the reason commits marked as new is because i did a bit of clean up today before making this PR and basically rewritten commit history in my fork.

nrader95 committed:
IsAliveVersion default entity struct crash fix

alkboda commented:
I have no great desire to rewrite all my systems classes constructors to manually define false useBuffer parameter. Wiki clearly states rules how to work with entities within basic system ancestors Update method. See no reason to use worse performance case as default behavior. If I need safe context, then I set useBuffer manually. But that's my personal opinion.

nrader95 commented:
The thing with no `useBuffer` param is that when it does go wrong, trying to detect it is a real pain. So chances are, you wont even know you *need* this safe context.
What's more, the potential perf gain doesnt even matter most of the time, because your system has to have a very light update to care about such overhead and do so in single-thread (useBuffer is not used in multi-thread, after all), on a great number of entities. That's the only situation where you might reasonably want to disable the buffering, but such systems are not that common in my experience. Unless you trying to write game engine of your own or something, with DefaultECS as the basis.

alkboda commented:
Debugging ECS is really painful, totally agree on this. But buffering systems updates isn't only thing that matters. We're developed a bunch of instruments for debugging, and each time we're forced to think out something new(eg. last time I wrote wrapper for ReanOnlySpan which I return from EntitySet::GetEntities() and which tracks changes made to EntitySet while ReadOnlySpan is enumerated - that solves all my problems with incorrect Updates within my systems).
UseBuffer situations could be detected by switching default parameter value to true - if that solves the problem, then definitely some of your systems violates rules.

nrader95 committed:
Drop .NetStandard 1.1 support

nrader95 committed:
Thread-based DefaultParallelRunner

nrader95 committed:
Entity misuse runtime detection(and its on/off usage property) for debug configurations

nrader95 committed:
Help update

nrader95 commented:
@Doraku So i take it this repo is abandoned since there is no reply after a whole year?

nrader95 commented:
@Doraku So, any response?
Since you seem to be active again.

Doraku commented:
On the debug stuff, I'll probably opt for a debug package or the nuget containing both release and debug dll and reference one or the other based on the configuration 🤔 I like the idea of better error reporting during development.

On the buffer default, I think it should be possible to keep it to false while still checking for incorrect usage in debug with extra code executing in the system so people are clearly notified of what they did wrong instead of just guessing like currently.

nrader95 committed:
Exclude tmp dotnet folders (bin/obj) for vscode

nrader95 committed:
Fix IDE Errors
-Removed old frameworks from Tests csproj
-Removed redundant ; char in frameworks section of main csproj

nrader95 commented:
I guess there is not much reason to bother with this PR any longer, since you dont seem to be interested.
